backends/imgui_impl_xlib.cpp

```diff
@@ -0,0 +1,644 @@
+
+            if (req->target == bd->XA_TARGETS)
+            {
+                XChangeProperty(bd->Dpy, req->requestor, req->property,
+                                    4 /* XA_ATOM */, 32, PropModeReplace,
+                                    (unsigned char*)bd->XA_MIME, 6);
+                sevent.xselection.property = req->property;
+                sevent.xselection.target = bd->XA_TARGETS;
+            }
+            else if (bd->ClipboardTextData)
+            {
+                for (unsigned int i = 0; i < sizeof(text_mime_types)/sizeof(text_mime_types[0]); i++)
+                {
+                    if (req->target != bd->XA_MIME[i])
+                        continue;
+
+                    XChangeProperty(bd->Dpy, req->requestor, req->property,
+                        req->target, 8, PropModeReplace,
+                        (unsigned char*)bd->ClipboardTextData, strlen(bd->ClipboardTextData));
+                        sevent.xselection.property = req->property;
+                        sevent.xselection.target = req->target;
+                        
+                    break;
+                }
+            }
+
+            XSendEvent(bd->Dpy, req->requestor, 0, 0, &sevent);
+            XSync(bd->Dpy, False);
+            return true;
+        }
+        case SelectionClear:
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool ImGui_ImplXlib_Init(Display* display, Window window)
+{
+    ImGuiIO& io = ImGui::GetIO();
+    IM_ASSERT(io.BackendPlatformUserData == nullptr && "Already initialized a platform backend!");
+
+    // Setup backend capabilities flags
+    ImGui_ImplXlib_Data* bd = IM_NEW(ImGui_ImplXlib_Data)();
+    io.BackendPlatformUserData = (void*)bd;
+    io.BackendPlatformName = "imgui_impl_xlib";
+    io.BackendFlags |= ImGuiBackendFlags_HasMouseCursors;       // We can honor GetMouseCursor() values (optional)
+    io.BackendFlags |= ImGuiBackendFlags_HasSetMousePos;        // We can honor io.WantSetMousePos requests (optional, rarely used)
+
+    bd->Dpy = display;
+    bd->Win = window;
+
+    io.SetClipboardTextFn = ImGui_ImplXlib_SetClipboardText;
+    io.GetClipboardTextFn = ImGui_ImplXlib_GetClipboardText;
+    io.ClipboardUserData = nullptr;
+
+    // Store all used atoms
+    bd->XA_SELECTION = XInternAtom(bd->Dpy, "IMGUI_SELECTION", 0);
+    bd->XA_TARGETS = XInternAtom(bd->Dpy, "TARGETS", 0);
+    bd->XA_INCR = XInternAtom(bd->Dpy, "INCR", 0);
+    bd->XA_CLIPBOARD = XInternAtom(bd->Dpy, "CLIPBOARD", 0);
+
+    bd->MimeCount = sizeof(text_mime_types)/sizeof(text_mime_types[0]) + 1;
+    bd->XA_MIME = (Atom*)malloc(bd->MimeCount * sizeof(Atom));
+    bd->XA_MIME[0] = bd->XA_TARGETS;
+    for (unsigned int i = 1; i < bd->MimeCount; i++) {
+        bd->XA_MIME[i] = XInternAtom(bd->Dpy, text_mime_types[i-1], 0);
+    }
+
+    // Select keyboard/focus events
+    XSelectInput(display, window, KeyPressMask | KeyReleaseMask | FocusChangeMask);
```

Comment:
Would be nice to let the client application handle these or atleast document it, i just spent 2 hours debugging only to find out you overwrite the subscribed events 😀